### PR TITLE
Use new rtrim/ltrim to reduce usage of regex

### PR DESCRIPTION
**Pull request by alkino**

### CODE REVIEW

File: core/hoc/ConfigParser.hoc [hunk 1]
```diff
@@ -198,23 +198,12 @@ obfunc parseBlock() { localobj pendingBlock, configIn, fieldName, fieldValue
  * @param $s1 string to get trimmed
  */
 proc trimFront() {
-    strUtil.tail($s1, "[ \t]*", $s1)
+    strUtil.ltrim($s1, $s1);
```
Comment (jorblancoa) [inline core/hoc/ConfigParser.hoc:201]:
```suggestion
    strUtil.ltrim($s1, $s1)
```

File: core/hoc/TargetParser.hoc [hunk 2]
```diff
@@ -426,17 +426,15 @@ proc getTrimmedLine() { localobj targetFile, strobj
     //check for leading whitespace
     wscheck = tstr
     strobj.left( wscheck, 1 )
-    if( strcmp( wscheck, " " ) == 0 || strcmp( wscheck, "\t" ) == 0 ) {
-        ret = strobj.tail( tstr, "[ \t]+", tstr ) //strip leading white space
-    }
+    strobj.rtrim()
```
Comment (matz-e) [inline core/hoc/TargetParser.hoc:429]:
```suggestion
    strobj.ltrim()
```
_leading_ whitespace, right?

File: core/hoc/TargetParser.hoc [hunk 3]
```diff
@@ -426,17 +426,15 @@ proc getTrimmedLine() { localobj targetFile, strobj
...
 
     //strip comments
     if( strobj.substr(tstr, "#" ) != -1 ) {
         ret = strobj.head(tstr, "#", tstr )
     }
 
 	//trailing whitespace shouldn't be a problem, but 'trimmedLine' implies none at front or rear
-	// this doesn't hurt parsing, though, so I won't deal with it
+	strobj.ltrim()
```
Comment (matz-e) [inline core/hoc/TargetParser.hoc:437]:
```suggestion
	strobj.rtrim()
```

**Comment by iomaganaris** [comment]
FYI the failing tests in https://bbpgitlab.epfl.ch/hpc/sim/neurodamus/-/pipelines/167327 cannot be fixed because it's not possible to select a certain NEURON branch.
To test the `blueconfigs` CI with a certain NEURON branch you can add to the last commit message:
```
<commit_message>

NEURON_BRANCH=<your_branch>
```

**Pull request closed without merging**